src/vcpkg/dependencies.cpp

```diff
@@ -593,62 +592,59 @@ namespace vcpkg
-            const std::unordered_set<PackageSpec>& specs_as_set;
+            std::unordered_map<PackageSpec, std::vector<PackageSpec>> rev_edges;
 
-            RemoveAdjacencyProvider(const StatusParagraphs& status_db,
-                                    const std::vector<InstalledPackageView>& installed_ports,
-                                    const std::unordered_set<PackageSpec>& specs_as_set)
-                : status_db(status_db), installed_ports(installed_ports), specs_as_set(specs_as_set)
+            std::vector<PackageSpec> adjacency_list(const PackageSpec& spec) const override
             {
-            }
-
-            std::vector<PackageSpec> adjacency_list(const RemovePlanAction& plan) const override
-            {
-                if (plan.plan_type == RemovePlanType::NOT_INSTALLED)
+                auto it = rev_edges.find(spec);
+                if (it != rev_edges.end())
                 {
-                    return {};
+                    return it->second;
                 }
-
-                const PackageSpec& spec = plan.spec;
-                std::vector<PackageSpec> dependents;
-                for (auto&& ipv : installed_ports)
+                else
                 {
-                    auto deps = ipv.dependencies();
-
-                    if (std::find(deps.begin(), deps.end(), spec) == deps.end()) continue;
-
-                    dependents.push_back(ipv.spec());
+                    // not installed
+                    return {};
                 }
-
-                return dependents;
             }
 
-            RemovePlanAction load_vertex_data(const PackageSpec& spec) const override
-            {
-                const RequestType request_type = specs_as_set.find(spec) != specs_as_set.end()
-                                                     ? RequestType::USER_REQUESTED
-                                                     : RequestType::AUTO_SELECTED;
-                const StatusParagraphs::const_iterator it = status_db.find_installed(spec);
-                if (it == status_db.end())
-                {
-                    return RemovePlanAction{spec, RemovePlanType::NOT_INSTALLED, request_type};
-                }
-                return RemovePlanAction{spec, RemovePlanType::REMOVE, request_type};
-            }
+            PackageSpec load_vertex_data(const PackageSpec& s) const override { return s; }
 
             std::string to_string(const PackageSpec& spec) const override { return spec.to_string(); }
         };
 
-        auto installed_ports = get_installed_ports(status_db);
-        const std::unordered_set<PackageSpec> specs_as_set(specs.cbegin(), specs.cend());
-        return topological_sort(specs, RemoveAdjacencyProvider{status_db, installed_ports, specs_as_set}, {});
+        RemoveAdjacencyProvider p;
+        for (auto&& a : get_installed_ports(status_db))
+        {
+            p.rev_edges.emplace(a.spec(), std::initializer_list<PackageSpec>{});
```

Comment:
```suggestion
            p.rev_edges.emplace(a.spec());
```

I think you can do that

Reply:
I get errors on GCC when I make that change :(